app/Models/Song.php

```diff
@@ -180,17 +183,20 @@ public function updateSingle($title, $albumName, $artistName, $lyrics, $track, $
             // Not a compilation song
             $this->contributing_artist_id = null;
             $albumArtist = Artist::get($artistName);
-            $album = Album::get($albumArtist, $albumName, false);
+            $album = Album::get($albumArtist, $albumName, $year, false);
         } else {
             $contributingArtist = Artist::get($artistName);
             $this->contributing_artist_id = $contributingArtist->id;
-            $album = Album::get(Artist::getVarious(), $albumName, true);
+            $album = Album::get(Artist::getVarious(), $albumName, $year, true);
         }
 
+        $this->title = $title;
         $this->album_id = $album->id;
         $this->title = $title;
```

Comment:
@yawboateng Thanks for this catch, this was a merging error. Also, if you'd like to see these changes merged in, please let @phanan know. He'd like to see more people request these features before they get merged in. So the more people the want it the better chance it has to get merged in!